### PR TITLE
fix: Timing of deployment status and local run printouts

### PR DESCRIPTION
**Pull request by jonaro00**
## Description of change

1. ~Prints the "Starting ... on localhost:8000" string in local run earlier so that is always prints before any output from the run.~ Didn't make sense.
2. Wait 500ms before calling project summary after deploy. This delay fixes "Deployment has not entered running state" showing on normal deploys since the deploy now has enough time to enter Running/Crashed state.
3. Fixed alignment of some printouts in local run.
4. Other minor cleanup.

## How Has This Been Tested (if applicable)?

Tested on local run and deploy against prod.

### CODE REVIEW

File: cargo-shuttle/src/lib.rs [hunk 1]
```diff
@@ -439,12 +436,13 @@ impl Shuttle {
...
 
+        let service_name = self.ctx.project_name();
         let working_directory = self.ctx.working_directory();
 
         trace!("building project");
         println!(
-            "{:>12} {}",
-            "Building".bold().green(),
+            "{} {}",
+            "    Building".bold().green(),
```
Comment (chesedo) [inline cargo-shuttle/src/lib.rs:445]: I might be a bit confused, why is this change needed?

File: cargo-shuttle/src/lib.rs [hunk 2]
```diff
@@ -603,15 +590,27 @@ impl Shuttle {
...
         };
+        let addr = SocketAddr::new(addr.into(), run_args.port);
 
-        let addr = SocketAddr::new(addr, run_args.port);
+        println!(
+            "    {} {} on http://{}\n",
+            "Starting".bold().green(),
+            service_name,
+            addr
+        );
```
Comment (chesedo) [inline cargo-shuttle/src/lib.rs:613]:
```suggestion
        println!(
            "{:>12} {} on http://{}\n",
            "Starting".bold().green(),
            service_name,
            addr
        );
```
This just makes it easier to keep things aligned without having to count spaces.

**Review by chesedo — COMMENTED** [review]
Nice, thanks for the touch ups @jonaro00! I just have one question :smile:

File: cargo-shuttle/src/lib.rs (hunk 1 above)
Comment (jonaro00) [reply cargo-shuttle/src/lib.rs:445]: The bold and green modifiers add control characters before and after the word, meaning it is already more than 12 bytes/chars long when given to the formatter, meaning no spaces are added. (Try it yourself :D)

You can solve this by either adding the correct amount of space, or to add the space alignment before bold+green, resulting in 2 format-like calls.

I chose the former.

File: cargo-shuttle/src/lib.rs (hunk 2 above)
Comment (jonaro00) [reply cargo-shuttle/src/lib.rs:613]: Same as other comment

File: cargo-shuttle/src/lib.rs (hunk 1 above)
Comment (chesedo) [reply cargo-shuttle/src/lib.rs:445]: Ahh, of course :man_facepalming:

**Review by chesedo — APPROVED** [review]

**Pull request merged**